clients/src/test/java/org/apache/kafka/clients/producer/internals/SenderTest.java

```diff
@@ -567,6 +567,44 @@ public void testMetadataTopicExpiry() throws Exception {
         assertTrue(future.isDone(), "Request should be completed");
     }
 
+    @Test
+    public void senderThreadShouldNotGetStuckWhenThrottledAndAddingPartitionsToTxn() {
+        // We want MockClient#poll() to advance time so that eventually the backoff expires.
+        client.setAdvanceTimeDuringPoll(true);
```

Comment:
Could be replace with `this.time = new MockTime(1L);` (or some other auto-tick value)?

Reply:
I tried that. However, the problem is that the thing we want to test is not how many times we call `time.milliseconds()` but rather what values are passed into `NetworkClient#poll()`.

I wasn't able to find a way to use this strategy to get a test that failed on `trunk` and passed on my branch.